lib/libutee/tee_api.c

```diff
@@ -143,13 +162,28 @@ TEE_Result TEE_InvokeTACommand(TEE_TASessionHandle session,
 	TEE_Result res;
 	uint32_t ret_origin;
 	struct utee_params up;
-
+#ifdef CFG_TEE_BENCHMARK
+	uint32_t allocated = 0;
+	void *timebuf = NULL;
+
+	if (params) {
+		timebuf = TEE_Malloc(TEE_BENCH_RB_SIZE, TEE_MALLOC_FILL_ZERO);
+		if (!timebuf)
+			return TEE_ERROR_OUT_OF_MEMORY;
+		allocated = 1;
+		params[TEE_BENCH_DEF_PARAM].memref.buffer = timebuf;
+		params[TEE_BENCH_DEF_PARAM].memref.size = TEE_BENCH_RB_SIZE;
+	}
+#endif
 	__utee_from_param(&up, paramTypes, params);
 	res = utee_invoke_ta_command((uintptr_t)session,
 				      cancellationRequestTimeout,
 				      commandID, &up, &ret_origin);
 	__utee_to_param(params, NULL, &up);
-
+#ifdef CFG_TEE_BENCHMARK
+	if (allocated)
```

Comment:
BTW, just a little notice. Not sure how `CFG_NO_USER_MALLOC_GARBAGE` should affect deallocation logic (in [tee_user_mem.c](https://github.com/OP-TEE/optee_os/blob/master/lib/libutee/tee_user_mem.c)), but for defined `CFG_NO_USER_MALLOC_GARBAGE` `tee_user_mem_free` implementation just don't have any checks ( it just calls utee_free) if to compare with `tee_user_mem_free` implementation that are under `#else` section for `CFG_NO_USER_MALLOC_GARBAGE`, which clearly says that 
```	
       /* It is OK to free NULL */
	if (buffer == NULL || buffer == ARTIST)
		return;
```